client/src/pages/Admin/UserDetails.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <article>
+    <nav class="q-px-lg q-pt-md q-gutter-sm">
+      <q-breadcrumbs>
+        <q-breadcrumbs-el
+          label="All Users"
+          to="/admin/users"
+        />
+        <q-breadcrumbs-el label="User Details" />
+      </q-breadcrumbs>
+    </nav>
+    <div class="row justify-center q-px-lg">
+      <h2
+        class="col-sm-12"
+        data-cy="userDetailsHeading"
+      >
+        {{ user.username || `Loading...` }}
+      </h2>
+    </div>
+    <div class="row q-pa-lg q-col-gutter-lg">
+      <section class="col-sm-2 col-xs-12">
+        <div class="row justify-center">
+          <q-item-section
+            top
+            avatar
+            class="col-sm-12 col-xs-2 q-mb-lg"
+          >
+            <avatar-image
+              :user="user"
+              rounded
+              class="fit"
+            />
+          </q-item-section>
+        </div>
+      </section>
+      <section class="col-sm-10 col-xs-12">
+        <div class="row q-mb-sm">
+          <div class="col-3 q-pr-lg text-right text--grey hide--xxs">
+            Username
+          </div>
+          <div class="col">
+            <q-icon
+              name="person_outline"
+              class="text--grey"
+            />
+            {{ user.username }}
+          </div>
+        </div>
+        <div class="row q-mb-sm">
+          <div class="col-3 q-pr-lg text-right text--grey hide--xxs">
+            Email
+          </div>
+          <div class="col">
+            <q-icon
+              name="mail_outline"
+              class="text--grey"
+            />
+            {{ user.email }}
+          </div>
+        </div>
+        <div class="row q-mb-sm">
+          <div class="col-3 q-pr-lg text-right text--grey hide--xxs">
+            Display Name
+          </div>
+          <div class="col">
+            <div v-if="user.name">
+              <q-icon
+                name="label_outline"
+                class="text--grey"
+              />
+              {{ user.name }}
+            </div>
+            <div v-else>
+              <q-icon
+                name="o_do_disturb_on"
+                class="text--grey"
+              />
+              <span
+                class="text--grey text-weight-light"
+              >
+                No Display Name
+              </span>
+            </div>
+          </div>
+        </div>
+        <div class="row q-mt-lg">
+          <div class="col-3 q-pr-lg text-right text--grey">
+            Roles
+          </div>
+          <div class="col">
+            <div
+              v-for="role in user.roles"
+              :key="role.id"
+              class="text-weight-medium"
+            >
+              <q-icon
+                v-if="role.name === 'Application Administrator' || role.name === 'Publication Administrator'"
```

Comment:
I wonder if we should add a token field to roles table to return a 'token' like APPLICATION_ADMIN with the roles for programmatic matching like this rather than matching a human string that may change in the future if the name of a role changes. 

Not important for this issue but something to think about.

Reply:
100%